src/main/java/com/appdynamics/monitors/azure/AzureMonitorTask.java

```diff
@@ -8,120 +8,99 @@
-                        metricType = "last"; metricValue = currentDataNode.get("last").decimalValue();
-                    }
-                    else if (currentDataNode.has("maximum")){
-                        metricType = "maximum"; metricValue = currentDataNode.get("maximum").decimalValue();
+        for (Map<String, ?> resourceGroupFilter : resourceGroupFilters) {
+            List<Map<String, ?>> resourceTypeFilters = (List<Map<String, ?>>) resourceGroupFilter.get("resourceTypes");
+            for (Map<String, ?> resourceTypeFilter : resourceTypeFilters) {
+                List<Map<String, ?>> resourceFilters = (List<Map<String, ?>>) resourceTypeFilter.get("resources");
+                for (Map<String, ?> resourceFilter : resourceFilters) {
+                    String currentResourceGroupFilter = resourceGroupFilter.get("resourceGroup").toString();
+                    for (GenericResource resource : resources) {
+                        String currentResourceFilter = resourceFilter.get("resource").toString();
+                        String currentResourceTypeFilter = resourceTypeFilter.get("resourceType").toString();
+                        AzureMetrics azureMetricsTask = new AzureMetrics(
+                                resourceFilter,
+                                currentResourceGroupFilter,
+                                currentResourceFilter,
+                                currentResourceTypeFilter,
+                                resource,
+                                subscriptionName,
+                                countDownLatchAzure,
+                                metricWriteHelper,
+                                azure,
+                                configuration.getMetricPrefix());
+                        configuration.getContext().getExecutorService().execute("AzureMetrics", azureMetricsTask);
                     }
                 }
-                if (metricId != null &&
-                        metricNameValue != null &&
-                        metricType != null &&
-                        metricUnit != null &&
-                        metricValue != null){
-                    MetricPrinter metricPrinter = new MetricPrinter(configuration.getMetricWriter());
-                    metricPrinter.reportMetric(configuration.getMetricPrefix() +
-                                    metricId +
-                                    metricNameValue, metricValue);
-                }
             }
         }
-    }
 
-    private static String extractMetridId(String fullId){
-        String metricId;
-        String[] metricIdSegments = fullId.split("resourceGroups")[1].split("providers/Microsoft\\.");
-        metricId = metricIdSegments[0]+metricIdSegments[1];
-        metricId = metricId.replace("/","|");
-        return metricId;
+        try{
+            countDownLatchAzure.await();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```

Comment:
You should log the exception instead of printing the stack trace. 
`logger.error("Some Error Message", e); `